# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1441447662 -7200
#      Sat Sep 05 12:07:42 2015 +0200
# Branch stable
# Node ID 3be09ebf2105ec5a8d54d0f1c1cdbfa7ea722ac5
# Parent  7ddb10b311265174e6d7537687c75ab21a093238
do not leave debug modus when entering a command at debug prompt (bug #45737)

* octave-cmd.cc (octave_cmd_eval::execute): use command_editor::accept_line ()
  instead of command_editor::interrupt ()

diff --git a/libgui/src/octave-cmd.cc b/libgui/src/octave-cmd.cc
--- a/libgui/src/octave-cmd.cc
+++ b/libgui/src/octave-cmd.cc
@@ -73,10 +73,10 @@ octave_cmd_eval::execute ()
       // no valid identifier: use Fsource (), no debug possible
       Fsource (ovl (file_path));
       command_editor::replace_line ("");
     }
 
   command_editor::set_initial_input (pending_input);
   command_editor::redisplay ();
 
-  command_editor::interrupt ();
+  command_editor::accept_line ();
 }
